pkg/resourcemanager/controller/managedresource/reconciler.go

```diff
@@ -515,12 +517,27 @@ func (r *Reconciler) applyNewResources(ctx context.Context, log logr.Logger, ori
 				return err
 			}
 
-			if apierrors.IsInvalid(err) && operationResult == controllerutil.OperationResultUpdated && deleteOnInvalidUpdate(current) {
```

Comment:
Would it be possible to rather extend the `deleteOnInvalidUpdate` function instead of hard-coding the ConfigMap/Secret handling here?